internal/lsp/hover.go

```diff
@@ -2,25 +2,21 @@ package lsp
 
 import (
 	"github.com/hashicorp/hcl-lang/lang"
-	"github.com/hashicorp/terraform-ls/internal/mdplain"
-	"github.com/sourcegraph/go-lsp"
+	lsp "github.com/hashicorp/terraform-ls/internal/protocol"
 )
 
 func HoverData(data *lang.HoverData, cc lsp.TextDocumentClientCapabilities) lsp.Hover {
-	mdSupported := cc.Hover != nil &&
-		len(cc.Hover.ContentFormat) > 0 &&
+	mdSupported := len(cc.Hover.ContentFormat) > 0 &&
 		cc.Hover.ContentFormat[0] == "markdown"
 
-	value := data.Content.Value
-	if data.Content.Kind == lang.MarkdownKind && !mdSupported {
-		value = mdplain.Clean(value)
-	}
-
-	content := lsp.RawMarkedString(value)
-	rng := HCLRangeToLSP(data.Range)
+	// In theory we should be sending lsp.MarkedString (for old clients)
+	// when len(cc.Hover.ContentFormat) == 0, but that's not possible
+	// without changing lsp.Hover.Content field type to interface{}
+	//
+	// We choose to follow gopls' approach (i.e. cut off old clients).
```

Comment:
This was discussed briefly in https://github.com/microsoft/language-server-protocol/issues/1144#issuecomment-730321055